auth/oauth2/passport.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+var passport = require('passport');
+var request = require('request');
+var debug = require('debug')('portal:auth:oauth2');
+var jwt = require('jsonwebtoken');
+
+var OAuth2Strategy = require('passport-oauth').OAuth2Strategy;
+
+var utils = require('../../routes/utils');
+var portalGlobals = require('../../portalGlobals');
+
+var federate = require('../federate');
+
+var oauth2Strategy = null;
+
+if (portalGlobals.glob.auth.oauth2 && 
+    portalGlobals.glob.auth.oauth2.useOauth2) {
+    debug('Initializing Oauth2 Authentication');
+    var oauth2Glob = portalGlobals.glob.auth.oauth2;
+    
+    oauth2Strategy = new OAuth2Strategy({
+        authorizationURL: oauth2Glob.authorizationURL,
+        tokenURL: oauth2Glob.tokenURL,
+        clientID: oauth2Glob.clientID,
+        clientSecret: oauth2Glob.clientSecret,
+        callbackURL: oauth2Glob.callbackURL,
+        passReqToCallback: true
+    }, function(req, accessToken, refreshToken, profile, done) {
+        debug('Oauth2 Authentication');
+        var decodedProfile = jwt.decode(accessToken);
+        var defaultGroups = [];
+        if(decodedProfile['group']){
+          defaultGroups = decodedProfile['group'];
```

Comment:
I think it would make sense to use the `matchGroups` functionality from ADFS (which also uses the same strategy, the OAuth2 strategy).

Reply:
Can I change it in such a way that in case group information does not exist in the token ONLY in that case call /group call like adfs?

Reply:
Will do

Reply:
Just a question regarding this, so that I am clear on this. Do you wish to have same capability as ADFS, which is (OPTION A): (1) verify certificate (2) decode token (3) call groups endpoint to get back groups.  OR (OPTION B): you mean that I should call a  "matchGroup" function same as ADFS on the groups I retrieved from the token. In our case groups are part of token. I can implement OPTION A in such a way that it only calls groups end point in case it could not find groups in the token. Please advice.

Reply:
For now I will implement Option B,  will refactor later in case you think Option A is the right thing to do.

Reply:
IIRC the ADFS groups, which are also part of the JWT which is passed in (the token), are matched with the wicked groups; the ADFS connector supports mapping ADFS groups to wicked groups, and if you keep it as is for now, you would assume the wicked groups are named the same as the OAuth2 groups you are getting passed in. Or am I getting it wrong?

I can live with that for now, it will need to be adapted for wicked 1.0 anyway. So if you don't want to invest more, it's also fine, and we'll keep it like this.